### PR TITLE
osg pilot container probe (SOFTWARE-4169)

### PR DESCRIPTION
**Pull request by edquist**
lots of criticism expected at this point

### CODE REVIEW

**Comment by djw8605** [comment]
There is a lot of SQL stuff at the beginning.  Is there a possibility that we could use something like sqlalchemy?  That would make it more "python'y", and get rid of alot of the creates at the beginning of the file.

**Review by brianhlin — COMMENTED** [review]
Had some time to do a minor review

File: osg-pilot-container/osgpilot_meter [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+
+def query_current_attrs():
+    import htcondor
+    pool = 'flock.opensciencegrid.org' #:9618
+    coll = htcondor.Collector(pool)
+
+    filter_cond = 'SlotType != "Static"'
+    #filter_cond += ' && IsOsgVoContainer =?= True'
+
+    ads = coll.query(ad_type=htcondor.AdTypes.Startd, projection=want_attrs,
```
Comment (brianhlin) [inline osg-pilot-container/osgpilot_meter:250]: Could you make `want_attrs` and your other constants cap-case? I was initially confused as to where it was defined.

```suggestion
    ads = coll.query(ad_type=htcondor.AdTypes.Startd, projection=WANT_ATTRS,
```

File: osg-pilot-container/osgpilot_meter [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+    """
+    create view new_jobs as
+    select b.*
+--    from inbox b
+      from pslot_totals b
+      left join jobs a
+        on a.name = b.name
+     where a.name is null;
+    """,
+]
```
[the remark this answers is not in the document]
Comment (edquist) [reply osg-pilot-container/osgpilot_meter:237]: As discussed this morning, we are not going to worry about sqlalchemy for the first pass.

I'd be willing to look into doing a Quake Total Conversion for it, though i admit i am not at all familiar with sqlalchemy.

...

If you want to know how i really feel though, it's something like this:

>Carl, you get to write a new probe with sqlite!
>Your favorite, right?

> :smiley: 

>The only catch is you don't get to write any sql.

> :anguished:

File: rpm/gratia-probe.spec [hunk 3]
```diff
@@ -714,6 +715,25 @@ Gratia OSG accounting system probe for providing VM accounting.
...
+osg pilot container probe
+
+%files osg-pilot-container
+%defattr(-,root,root,-)
+%doc %{default_prefix}/gratia/slurm/README.html
+%dir %{default_prefix}/gratia/osg-pilot-container
+%{default_prefix}/gratia/osg-pilot-container/osgpilot_meter
+%{default_prefix}/gratia/osg-pilot-container/ProbeConfig
+%config(noreplace) %verify(not md5 size mtime) %{_sysconfdir}/gratia/osg-pilot-container/ProbeConfig
+%dir %{_localstatedir}/lib/gratia/osg-pilot-container
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply rpm/gratia-probe.spec:736]: > The rpm packaging is how we generate, eg, the ProbeConfig though.
>
> A docker container can certainly install the rpm!

We could also generate the ProbeConfig in the image build. Let's do that for now until we decide there's a need for the RPM.

**Comment by edquist** [comment]
@brianhlin, despite this PR being a big pile of hot garbage, how do you feel about us merging it in for now so we can get per commit docker builds going?

**Review by brianhlin — APPROVED** [review]
Let's do it!

**Pull request merged**